backend/sites/src/app/services/folio/folioContents.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { FolioContents } from '../../entities/folioContents.entity';
+import { Repository } from 'typeorm';
+import { FolioContentDTO } from '../../dto/folioContent';
+import { plainToInstance } from 'class-transformer';
+
+@Injectable()
+export class FolioContentsService {
+  constructor(
+    @InjectRepository(FolioContents)
+    private folioContentRepository: Repository<FolioContents>,
+  ) {}
+
+
+  async getSiteForFolio(folioId: string): Promise<FolioContents[]> {
+    try {
+      const sitesForFolioArr = await this.folioContentRepository.find({  
+        relations: ['site','folio'],
+        where: { folioId },
+      });
+      return sitesForFolioArr;
+    } catch (error) {
+      throw error;
+    }
+  }
+
+
+  async addFolioContent(inputDTO: FolioContentDTO): Promise<boolean> {
+    const { siteId, folioId } = inputDTO;
+    try {
+      const existingRecord = await this.folioContentRepository.findOne({
+        where: { siteId, folioId },
+      });
+
+      if (existingRecord) {
+        return false;
+      } else {
+        delete inputDTO.id;
+        const folioContent = plainToInstance(FolioContents, inputDTO); 
+         
+        const result = await this.folioContentRepository.save(folioContent);
+
+        if (result) {
+          return true;
+        }
+      }
+    } catch (error) {
+      throw new Error('Failed to folio content');
+    }
```

Comment:
Can we pls reword it to  `Failed to add folio content`?. I guess you must have meant the same but just missed add keyword :)

Reply:
done